src/dmd/access.d

```diff
@@ -28,163 +28,19 @@ import dmd.tokens;
-    {
-        if (smember.toParent() == dthis)
-            return true;
-        if (ClassDeclaration cdthis = dthis.isClassDeclaration())
-        {
-            foreach (b; *cdthis.baseclasses)
-            {
-                const Prot access = getAccess(b.sym, smember);
-                if (access.kind >= Prot.Kind.protected_ || isAccessible(smember, sfunc, b.sym, cdscope))
-                {
-                    return true;
-                }
-            }
-        }
-    }
-    else
-    {
-        if (smember.toParent() != dthis)
-        {
-            if (ClassDeclaration cdthis = dthis.isClassDeclaration())
-            {
-                foreach (b; *cdthis.baseclasses)
-                    if (isAccessible(smember, sfunc, b.sym, cdscope))
-                        return true;
-            }
-        }
-    }
-    return false;
-}
-
 /*******************************
  * Do access check for member of this class, this class being the
  * type of the 'this' pointer used to access smember.
  * Returns true if the member is not accessible.
  */
 bool checkAccess(AggregateDeclaration ad, Loc loc, Scope* sc, Dsymbol smember)
 {
-    FuncDeclaration f = sc.func;
-    AggregateDeclaration cdscope = sc.getStructClassScope();
     static if (LOG)
     {
         printf("AggregateDeclaration::checkAccess() for %s.%s in function %s() in scope %s\n", ad.toChars(), smember.toChars(), f ? f.toChars() : null, cdscope ? cdscope.toChars() : null);
     }
-    Dsymbol smemberparent = smember.toParent();
-    if (!smemberparent || !smemberparent.isAggregateDeclaration())
-    {
-        static if (LOG)
-        {
-            printf("not an aggregate member\n");
-        }
-        return false; // then it is accessible
-    }
-    // BUG: should enable this check
-    //assert(smember.parent.isBaseOf(this, NULL));
-    bool result;
-    Prot access;
-    if (smemberparent == ad)
-    {
-        access = smember.prot();
-        result = access.kind >= Prot.Kind.public_ || hasPrivateAccess(ad, f) || isFriendOf(ad, cdscope) || (access.kind == Prot.Kind.package_ && hasPackageAccess(sc, smember)) || ad.getAccessModule() == sc._module;
-        static if (LOG)
-        {
-            printf("result1 = %d\n", result);
-        }
-    }
-    else if ((access = getAccess(ad, smember)).kind >= Prot.Kind.public_)
-    {
-        result = true;
-        static if (LOG)
-        {
-            printf("result2 = %d\n", result);
-        }
-    }
-    else if (access.kind == Prot.Kind.package_ && hasPackageAccess(sc, ad))
-    {
-        result = true;
-        static if (LOG)
-        {
-            printf("result3 = %d\n", result);
-        }
-    }
-    else
-    {
-        result = isAccessible(smember, f, ad, cdscope);
-        static if (LOG)
-        {
-            printf("result4 = %d\n", result);
-        }
-    }
-    if (!result && (!(sc.flags & SCOPE.onlysafeaccess) || sc.func.setUnsafe()))
+    
```

Comment:
there seems to be some trailing whitespace here tha's causing the autotester failure.